refapp/SKWebApi/Skills/InfiniteChatSkill.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System.Globalization;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.CoreSkills;
+using Microsoft.SemanticKernel.KernelExtensions;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace SKWebApi.Skills;
+
+public class InfiniteChatSkill
+{
+    private readonly IKernel _kernel;
+
+    public InfiniteChatSkill(IKernel kernel)
+    {
+        this._kernel = kernel;
+    }
+
+    [SKFunction("Extract user intent")]
+    [SKFunctionName("ExtractUserIntent")]
+    [SKFunctionContextParameter(Name = "audience", Description = "The audience the chat bot is interacting with.")]
+    public async Task<string> ExtractUserIntentAsync(SKContext context)
+    {
+        var tokenLimit = SystemPromptDefaults.completionTokenLimit;
+        var historyTokenBudget =
+            tokenLimit -
+            SystemPromptDefaults.responseTokenLimit -
+            this.EstimateTokenCount(string.Join("\n", new string[]{
+                SystemPromptDefaults.systemDescriptionPrompt,
+                SystemPromptDefaults.systemIntentPrompt,
+                SystemPromptDefaults.systemIntentContinuationPrompt})
+            );
+
+        var intentExtractionVariables = new ContextVariables();
+        intentExtractionVariables.Set("tokenLimit", historyTokenBudget.ToString(new NumberFormatInfo()));
+        intentExtractionVariables.Set("knowledgeCutoff", SystemPromptDefaults.knowledgeCutoffDate);
+        intentExtractionVariables.Set("audience", context["audience"]);
+
+        var completionFunction = this._kernel.CreateSemanticFunction(
+            SystemPromptDefaults.systemIntentExtractionPrompt,
+            skillName: nameof(InfiniteChatSkill),
+            description: "Complete the prompt.");
```

Comment:
Isn't the description "Extract user intent"?

Reply:
Extracting the intent is basically completing the prompt, which instructs the model what to perform.